### PR TITLE
supporting Redis 4

### PR DESCRIPTION
**Pull request by oleurud**
Solving a part of CartoDB/Windshaft-cartodb#819
Main PR CartoDB/Windshaft-cartodb#849

Updating node-redis package adding Redis 4 support.

Of course, this change must be work with Redis 3.

`npm` have changed the `package.json` format

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -134,7 +141,7 @@ function makePool(options, database) {
                     callbackCalled = true;
                     callback(err, client);
                 }
-                client.end();
+                client.end(true);
```
Comment (dgaubert) [inline index.js]: Please, give a name to this boolean param

File: index.js [hunk 2]
```diff
@@ -144,12 +151,12 @@ function makePool(options, database) {
...
                     }
                 });
-            })
+            });
         },
 
         destroy: function(client) {
             client.quit();
-            client.end();
+            client.end(true);
```
Comment (dgaubert) [inline index.js]: Same here

**Comment by oleurud** [comment]
About the new parameter in client.end method --> https://github.com/NodeRedis/node_redis#clientendflush

```
Forcibly close the connection to the Redis server. Note that this does not wait until all replies have been parsed. If you want to exit cleanly, call client.quit() as mentioned above.

You should set flush to true, if you are not absolutely sure you do not care about any other commands. If you set flush to false all still running commands will silently fail.

client.end() without the flush parameter set to true should NOT be used in production!
```

**Comment by dgaubert** [comment]
So, updating to redis 4 we need to call `.end()` with the  param set to true, rigth?

If so, we need to add that param here: https://github.com/CartoDB/node-redis-mpool/blob/eef21555cec4266d6eb727da43ea412974a68b09/index.js#L137 and anywhere we use `.end()`

**Pull request merged**